spec/factories/saml_attributes.rb

```diff
@@ -0,0 +1,466 @@
+# frozen_string_literal: true
+
+# rubocop:disable Layout/LineLength
+FactoryBot.define do
+  factory :idme_loa1, class: OneLogin::RubySaml::Attributes do
+    transient do
+      authn_context { 'http://idmanagement.gov/ns/assurance/loa/1/vets' }
+    end
+    uuid { ['0e1bb5723d7c4f0686f46ca4505642ad'] }
+    email { ['kam+tristanmhv@adhocteam.us'] }
+    multifactor { [false] }
+    level_of_assurance { ['1'] }
+
+    initialize_with { new(attributes.stringify_keys) }
+  end
+
+  factory :idme_loa3, class: OneLogin::RubySaml::Attributes do
+    transient do
+      authn_context { 'http://idmanagement.gov/ns/assurance/loa/3/vets' }
+    end
+    uuid { ['0e1bb5723d7c4f0686f46ca4505642ad'] }
+    email { ['kam+tristanmhv@adhocteam.us'] }
+    fname { ['Tristan'] }
+    lname { ['MHV'] }
+    mname { [''] }
+    social { ['111223333'] }
+    gender { ['male'] }
+    birth_date { ['1735-10-30'] }
+    multifactor { [true] } # always true for these types
+    level_of_assurance { ['3'] }
+
+    initialize_with { new(attributes.stringify_keys) }
+  end
+
+  factory :mhv_basic, class: OneLogin::RubySaml::Attributes do
+    transient do
+      authn_context { 'myhealthevet' }
+    end
+    mhv_icn { [''] }
+    mhv_profile { ['{"accountType":"Basic"}'] }
+    mhv_uuid { ['12345748'] }
+    email { ['kam+tristanmhv@adhocteam.us'] }
+    multifactor { [false] }
+    uuid { ['0e1bb5723d7c4f0686f46ca4505642ad'] }
+    level_of_assurance { ['0'] } # TODO: check this is idme_loa
+
+    initialize_with { new(attributes.stringify_keys) }
+  end
+
+  factory :mhv_advanced, class: OneLogin::RubySaml::Attributes do
+    transient do
+      authn_context { 'myhealthevet' }
+    end
+    mhv_icn { ['1012853550V207686'] }
+    mhv_profile { ['{"accountType":"Advanced"}'] }
+    mhv_uuid { ['12345748'] }
+    email { ['kam+tristanmhv@adhocteam.us'] }
+    multifactor { [false] }
+    uuid { ['0e1bb5723d7c4f0686f46ca4505642ad'] }
+    level_of_assurance { ['0'] } # TODO: check this is idme_loa
+
+    initialize_with { new(attributes.stringify_keys) }
+  end
+
+  factory :mhv_premium, class: OneLogin::RubySaml::Attributes do
+    transient do
+      authn_context { 'myhealthevet' }
+    end
+    mhv_icn { ['1012853550V207686'] }
+    mhv_profile {
+      [
+        '{"accountType":"Premium","availableServices":{"21":"VA Medications","4":"Secure Messaging","3":"VA Allergies","2":"Rx Refill","12":"Blue Button (all VA data)","1":"Blue Button self entered data.","11":"Blue Button (DoD) Military Service Information"}}'
```

Comment:
Can we pretty this up with some line breaks for readability?

Reply:
Done